File: backend/src/main/java/com/github/wcosmedlr/dao/Expense.java
package com.github.wcosmedlr.dao;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.Date;

@Entity
public class Expense {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String description;
    private Date timeStamp;
    @ManyToOne
    private Member owner;
    private Double value;

    public void setId(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getTimeStamp() {
        return timeStamp;
    }

    public void setTimeStamp(Date timeStamp) {
        this.timeStamp = timeStamp;
    }

    public Member getOwner() {
        return owner;
    }

    public void setOwner(Member owner) {
        this.owner = owner;
    }

    public Double getValue() {
        return value;
    }

    public void setValue(Double value) {
        this.value = value;
    }
}

File: frontend/src/models/expense.ts
import { Balance} from "./balance";
import { buildMember } from "./member";
import { MoneyUnit } from "./moneyunit";

export interface Expense extends MoneyUnit {
  id?: number;
  description: string;
  timeStamp: Date;
}

export const buildExpense = ({
  id = undefined,
  description = '10',
  timeStamp = new Date(),
  owner = buildMember({}),
  value = 0
}: Partial<Expense>): Expense => ({
  id,
  description,
  timeStamp,
  owner,
  value,
});